api/v1beta1/azuremachine_types.go

```diff
@@ -133,6 +134,10 @@ type SpotVMOptions struct {
 	// MaxPrice defines the maximum price the user is willing to pay for Spot VM instances
 	// +optional
 	MaxPrice *resource.Quantity `json:"maxPrice,omitempty"`
+
+	// EvictionPolicy defines the behavior of the virtual machine when it is evicted. It can be either Delete or Deallocate.
+	// +optional
+	EvictionPolicy *compute.VirtualMachineEvictionPolicyTypes `json:"evictionPolicy,omitempty"`
```

Comment:
we want to avoid adding a dependency on the Azure SDK in the api package. This is so that the SDK can we swapped out easily for another (for example this might be applicable to support other clouds besides Azure public cloud). Here instead of using `compute.VirtualMachineEvictionPolicyTypes` you can define a new type